src/k_retrieval.py

```diff
@@ -0,0 +1,31 @@
+def k_retrieval(distances_list, similarity_methods, k):
+    
+    """
+    
+    Returns the K more similar images of the database from a list of distance obj
+    
+    """
+    
+    method=distances_list[0].method
+
+    if method == "euclidean" or method == "L1_dist" or method == "x2_dist" or method == "hellinger":
```

Comment:
The `if` and the `else` have almost the same code. I would change to something like 
```suggestion
    maximization_required = method not in [ "euclidean", "L1_dist", "x2_dist", "hellinger"]
    distances_list.sort(key=lambda x: x.distance, reverse=maximization_required)
    k_distances=distances_list[0 : k+1]
    

```

Reply:
Even better, I would store in the "distance" class if the metric needs maximization or minimization.

Reply:
Yep, you're right, I'm changing it.